lib/puppet/provider/cisco_bgp_af/nxapi.rb

```diff
@@ -136,6 +149,96 @@ def properties_set(new_af=false)
       @af.send("#{prop}=", @property_flush[prop]) if
         @af.respond_to?("#{prop}=")
     end
+    # Custom set methods
+    dampening_set
+  end
+
+  # Property 'dampening' helper and custom setter methods
+  def dampening_enable
+    @af.dampening = '' unless dampening_properties?
+  end
+
+  def dampening_disable
+    @af.dampening = nil
+    @property_flush[:dampening_half_time] = nil
+    @property_flush[:dampening_reuse_time] = nil
+    @property_flush[:dampening_suppress_time] = nil
+    @property_flush[:dampening_max_suppress_time] = nil
+    @property_flush[:dampening_dampening_routemap] = nil
+  end
+
+  def dampening_properties?
+    @property_flush[:dampening_half_time] ||
+      @property_flush[:dampening_reuse_time] ||
+      @property_flush[:dampening_suppress_time] ||
+      @property_flush[:dampening_max_suppress_time] ||
+      @property_flush[:dampening_routemap]
+  end
+
+  def dampening_properties_set
+    if @property_flush[:dampening_half_time]
+      half = @property_flush[:dampening_half_time]
+    else
+      half = @af.dampening_half_time
+    end
+
+    if @property_flush[:dampening_reuse_time]
+      reuse = @property_flush[:dampening_reuse_time]
+    else
+      reuse = @af.dampening_reuse_time
+    end
+
+    if @property_flush[:dampening_suppress_time]
+      suppress = @property_flush[:dampening_suppress_time]
+    else
+      suppress = @af.dampening_suppress_time
+    end
+
+    if @property_flush[:dampening_max_suppress_time]
+      max_suppress = @property_flush[:dampening_max_suppress_time]
+    else
+      max_suppress = @af.dampening_max_suppress_time
+    end
+    @af.dampening = [half, reuse, suppress, max_suppress]
+  end
+
+  def dampening_routemap_set
+    if @property_flush[:dampening_routemap]
+      rtmap = @property_flush[:dampening_routemap]
+    else
+      rtmap = @af.dampening_routemap
+    end
+
+    @af.dampening = rtmap
+  end
+
+  def dampening_set
+    dampening_disable if @property_flush[:dampening_state].to_s == 'false'
+    dampening_enable if @property_flush[:dampening_state].to_s == 'true'
+
+    return if @resource[:dampening_state].to_s == 'false'
+    return unless dampening_properties?
```

Comment:
Shouldn't this be the first check? If there are no dampening props in flush then you should be done.